pytensor/tensor/rewriting/linalg.py

```diff
@@ -855,6 +857,59 @@ def rewrite_det_kronecker(fgraph, node):
     return [det_final]
 
 
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_solve_kron_to_solve(fgraph, node):
+    """
+    Given a linear system of the form:
+
+    .. math::
+
+        (A \\otimes B) x = y
+
+    Define :math:`\text{vec}(x)` as a column-wise raveling operation (``x.reshape(-1, order='F')`` in code). Further,
+     define :math:`y = \text{vec}(Y)`. Then the above expression can be rewritten as:
+
+    .. math::
+
+        x = \text{vec}(B^{-1} Y A^{-T})
+
+    Eliminating the kronecker product from the expression.
+    """
+
+    if not isinstance(node.op.core_op, SolveBase):
+        return
+
+    solve_op = node.op
+    props_dict = solve_op.core_op._props_dict()
+
+    if props_dict["b_ndim"] != 1:
```

Comment:
b_ndim is just a batched vector solve, could you move it to a batch dimension of b?

```python
import numpy as np
import pytensor.tensor as pt

a = pt.matrix("a", shape=(3, 3))
b = pt.matrix("b", shape=(3, 5))
out1 = pt.linalg.solve(a, b, b_ndim=2)
out2 = pt.moveaxis(pt.linalg.solve(a, pt.moveaxis(b, -1, 0), b_ndim=1), 0, -1)

rng = np.random.default_rng(20)
test_dict = {
    a: rng.normal(size=(a.type.shape)),
    b: rng.normal(size=(b.type.shape))
}
np.testing.assert_allclose(out1.eval(test_dict), out2.eval(test_dict))
```